header-rewrite-filter/utility.cc

```diff
@@ -63,6 +73,67 @@ bool requiresArgument(MatchType match_type) {
     return (match_type == MatchType::Exact || match_type == MatchType::Substr || match_type == MatchType::Prefix);
 }
 
+std::tuple<absl::Status, std::string> getHeaderValue(Http::RequestOrResponseHeaderMap& headers, std::vector<absl::string_view> arguments) {
```

Comment:
It's not immediately clear why this takes in a vector of string views. I think the interface should be something like `getHeaderValue(headerMap, headerKey, position)`. The caller can provide `-1` if the parsed operation doesn't have an explicit position

Reply:
The only thing is that I was validating the arguments to the function within `getHeaderValue` and `getUrlp`, which now I have to do outside, which will be repeated code every time a processor needs to call these functions.